main.go

```diff
@@ -57,25 +59,27 @@ func readConfig() config {
 	return conf
 }
 
-func startHttpsServer(simpleServer httpd.SimpleServer) error {
-	errs := make(chan error, 1)
-	simpleServer.Start(errs)
+func startHttpServerAndWait(simpleServer httpd.SimpleServer, callback func()) error {
```

Comment:
@jcchavezs see if this implementation make sense. I combined start and wait into a single method. After starting, either I listen to errors or os signal to terminate - on either one, I come out of the blocking select and return.